deploy/crd.yaml

```diff
@@ -11,9 +11,357 @@ spec:
     singular: flinkapplication
     shortNames:
       - flinkapp
-
   scope: Namespaced
   version: v1alpha1
+  validation:
+    # openAPIV3Schema is the schema for validating custom objects.
+    openAPIV3Schema:
+      type: object
+      properties:
+        spec:
+          type: object
+          properties:
+            image:
+              type: string
+            imagePullPolicy:
+              type: string
+              enum: [Always, Never, IfNotPresent]
+            imagePullSecrets:
+              type: array
+              items:
+                type: object
+                properties:
+                  name:
+                    type: string
+            jarName:
+              type: string
+            programArgs:
+              type: string
+            entryClass:
+              type: string
+            flinkVersion:
+              type: string
+            restartNonce:
+              type: string
+            parallelism:
+              type: integer
+              minimum: 1
+            deleteMode:
+              type: string
+              enum: [Savepoint, None, ForceCancel]
+            deploymentMode:
+              type: string
+              enum: [Single, Dual]
```

Comment:
Since Single is not currently supported, should we remove?

Reply:
Sure